### PR TITLE
Documentation generation and credential handling fixes

### PR DESCRIPTION
**Pull request by BrianFarnhill**
Minor bug fixes also included as well as the set up for the documentation generation

### CODE REVIEW

**Comment by BrianFarnhill** [comment]
@caadam - Another big code review bloke. I'm still testing the search resources on my multi-server setups to confirm they play nice there (they are solid on single server labs) but I thought I would get the PR going and tweak it as I test things. I'll give you a heads up when anything changes.

File: Modules/xSharePoint/DSCResources/MSFT_xSPSearchRoles/MSFT_xSPSearchRoles.psm1 [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+        $ConfirmPreference = 'None'
+
+        $AdminExists = $false;
+        $CrawlerExists = $false;
+        $ContentProcessingExists = $false;
+        $AnalyticsProcessingExists = $false;
+        $QueryProcessingExists = $false;
+
+        $ssi = Get-SPEnterpriseSearchServiceInstance -Identity $env:COMPUTERNAME 
+        $ssa = Get-SPEnterpriseSearchServiceApplication
```
Comment (camiloborges) [inline Modules/xSharePoint/DSCResources/MSFT_xSPSearchRoles/MSFT_xSPSearchRoles.psm1]: What if you have multiple SSAs? Shouldn't you use the $ServiceAppName"?

File: Modules/xSharePoint/DSCResources/MSFT_xSPSearchRoles/MSFT_xSPSearchRoles.psm1 [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+            Start-SPEnterpriseSearchServiceInstance -Identity $ssi
+        }
+
+        #Wait for Search Service Instance to come online
+        $online = Get-SPEnterpriseSearchServiceInstance -Identity $ssi; 
+        do {
+            $online = Get-SPEnterpriseSearchServiceInstance -Identity $ssi; 
+            Write-Verbose "Waiting for service: $($online.TypeName)"
+        } 
+        until ($online.Status -eq "Online")
```
Comment (camiloborges) [inline Modules/xSharePoint/DSCResources/MSFT_xSPSearchRoles/MSFT_xSPSearchRoles.psm1]: Search services doesn't usually get stuck, but, I think you need to have a timeout here, don't you?

File: Modules/xSharePoint/DSCResources/MSFT_xSPSearchRoles/MSFT_xSPSearchRoles.psm1 [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+                                                               -SearchServiceInstance $ssi
+            }
+
+            $IndexComponent1 = Get-SPEnterpriseSearchComponent -SearchTopology $newTopology | Where-Object {
+                ($_.GetType().Name -ccontains "IndexComponent")
+            }
+            
+            #Add the First search index if no other indexes already exist required
+            if($null -eq $IndexComponent1) {
+                Write-Verbose "Adding First search indedx at partition $($params.FirstPartitionIndex)"
```
Comment (camiloborges) [inline Modules/xSharePoint/DSCResources/MSFT_xSPSearchRoles/MSFT_xSPSearchRoles.psm1]: Index :) (Indedx)

**Comment by BrianFarnhill** [comment]
@camiloborges Fixed the following based on your feedback

 - Now both search roles and search index use SSA name to look up the SSA
 - There is a loop count to hang off the starting of the search service app. Hard coded to 10 minutes
 - Fixed the type of index in the log message

**Comment by camiloborges** [comment]
cool. I will try to run few scenarios in a multi server farm. I should have
a reasonably up to date farm in the office.

**Comment by BrianFarnhill** [comment]
@camiloborges  I'm seeing an issue at the moment where the folder for the search indexes isn't being created on separate remote servers. Say I have sharepoint1 running the search roles and search index partition resources, telling them to create index partitions 0 and 1 on sharepoint3 and sharepoint4. I'll see that SharePoint3 gets a folder created for the indexes locally, but for some reason sharepoint 4 won't. That's the last bug I've found with it, the rest of the topology components are created fine.

**Comment by BrianFarnhill** [comment]
@camiloborges @caadam Hey guys, don't bother reviewing the search topology stuff, I'm going to pull it out and push it to the next release. Turns out there was a couple of fatal flaws in my approach so I'm going back to the drawing board on this one. I'll remove the files from this PR later tonight and get the rest of the changes pushed through though

**Pull request merged**